files/js/setpassword.js

```diff
@@ -0,0 +1,30 @@
+jQuery(document).ready(function($) {
+    // Get translated strings from JSON object.
+    var i18n = JSON.parse(document.getElementById('adminSetPasswordData').innerHTML);
+    // Allow admin to set password directly.
+    // Include input for new password.
+    var frm = $('form[action="manage_user_reset.php"]');
+    $('<div class="adminsetpassword" style="position:absolute;margin:40px 0 0 0;width:100%;"><input type="password" id="new_passw" class="input-sm" /><input type="submit" value="'+i18n.change_password+'" class="btn btn-primary btn-white btn-round" id="submit_newpassw" /></div>').prependTo(frm);
+
+    $('#submit_newpassw').on('click', function() {
+        var sel = $('#new_passw');
+
+        if (sel.val().length > 5) {
+            $.post(
+            'plugins/AdminSetPassword/pages/ajax.php',
```

Comment:
This does not work for me (I get a 403 error) on the XHR, and it would probably fail on many other MantisBT sites too, depending on web server config.

The reason is that MantisBT comes by default with a *.htaccess* file that prevents direct access to anything below `/plugins` directory.

The clean way to execute the AJAX this would be by defining a REST route for the POST request (see [Announce ](https://github.com/mantisbt-plugins/Announce/blob/master/Announce.php) plugin for an example, lines 46, 135, 160). If you do this, you'll need to bump the minimum requirement to MantisBT core 2.3.0.

Alternatively, maybe it could work if you proxy ajax.php page via plugin.php like so `/plugin.php?page=AdminSetPassword/ajax` but never tried that myself (and TBH it would be a bit of a hack).